src/meetingsession/DefaultMeetingSession.ts

```diff
@@ -116,13 +129,45 @@ export default class DefaultMeetingSession implements MeetingSession, Destroyabl
     if (isDestroyable(this.audioVideoController)) {
       await this.audioVideoController.destroy();
     }
+    if (isDestroyable(this.eventReporter)) {
+      await this.eventReporter.destroy();
+    }
 
     this._logger = undefined;
     this._configuration = undefined;
     this._deviceController = undefined;
     this.audioVideoFacade = undefined;
     this.audioVideoController = undefined;
     this.contentShareController = undefined;
+    this._eventReporter = undefined;
+  }
+
+  private setupEventReporter(
+    configuration: MeetingSessionConfiguration,
+    logger: Logger,
+    eventReporter?: EventReporter
+  ): void {
+    if (eventReporter) {
+      this._eventReporter = eventReporter;
+    } else {
+      const eventsIngestionURL = configuration.urls.eventsIngestionURL;
+      if (eventsIngestionURL) {
+        const {
+          meetingId,
+          credentials: { attendeeId, joinToken },
+        } = configuration;
+        const meetingEventsClientConfiguration: EventsClientConfiguration = new MeetingEventsClientConfiguration(
+          meetingId,
```

Comment:
nit: Can we just use configuration.meetingId, configuration.credentials.atendeeId, etc. here? Don't think we need the above line.

Reply:
Here it avoids actually three lines and repetitive `configuration.credentials` for both the `joinToken` and `attendeeId`. Also avoids `configuration.` few times once for `meetingId` and twice for `credentials`.

Reply:
Also, I had it earlier in normal where I got comment that I can use de-structring here.